lambda_src/drivers/destination_s3.py

```diff
@@ -101,13 +100,19 @@ def write(
     row_index: int,
 ) -> Dict[Text, Any]:
     bucket, prefix = parse_destination_uri(destination)
-    encoded_datum = (
-        '\n'.join(json.dumps(d) for d in datum)
-        if isinstance(datum, list)
-        else json.dumps(datum, default=str)
-    )
+    if isinstance(datum, bytes):
+        encoded_datum = datum
+        prefixed_filename = hash_format(
+            prefix, encoded_datum, sha2=lambda x: sha256(x).hexdigest()
+        )
+    else:
+        encoded_datum = (
+            '\n'.join(json.dumps(d) for d in datum)
+            if isinstance(datum, list)
+            else json.dumps(datum, default=str)
+        )
+        prefixed_filename = f'{prefix}{batch_id}_row_{row_index}.data.json'
```

Comment:
We'd want to maintain uniformity by hash-formatting in either of the above cases right?

Reply:
it would be nice to have, but not necessary for this use-case, so can also wait

Reply:
Makes sense 👍